### DISCUSSION TITLE
EffectiveCost missing values after deploying .7

### DISCUSSION THREAD

**Discussion by lundejd** [microsoft/finops-toolkit#1229] [Q&A] [answered]
I finally got around to upgrading to .7 and ran some tests with my .6 Power BI files and noticed many of my measures were not showing up. Upon further investigation I noticed I have virtually no EffectiveCost amounts. (I'd say none but just a couple days in December had a few hundred dollars where I should be seeing hundreds of thousands). 

Figuring it might be my Power BI customizations I opened up an untouched version of CostSummary.storage.pbit and tried again - same results.

This led me to deploying a completely new .7 in a new resource group thinking maybe part of the upgrade did something but this morning after re-loading I am still seeing the exact same thing.

My .6 deployments with similar export jobs are all still showing just fine, so it doesn't seem to be in the underlying data we're getting from Microsoft. I also tried forcing an overwrite of the months data with no change in result. 

I did some searching and noticed the (below) from the .7 update blog so I dug around and found this line from the CostDetails query that seems to be the culprit as its replacing nearly everything with $0

`// Fix savings plan purchases
            each [EffectiveCost], each if [ChargeCategory] = "Purchase" and ([CommitmentDiscountId] = "" or [CommitmentDiscountId] = null) then [EffectiveCost] else 0.0, Replacer.ReplaceValue, {"EffectiveCost"}),`

Seems like this logic would say when [ChargeCategory] = "Usage" which most of ours is, it would be 0.0. 

I replaced the line with the following which seems like it may have fixed things (re-running the numbers to validate).

`// Fix savings plan purchases
            each [EffectiveCost], each if [ChargeCategory] <> "Purchase" then [EffectiveCost] else if [ChargeCategory] = "Purchase" and ([CommitmentDiscountId] = "" or [CommitmentDiscountId] = null) then [EffectiveCost] else 0.0, Replacer.ReplaceValue, {"EffectiveCost"}),`

.7 has been out a while, seems odd this hasn't been found by now but maybe my searching jus hasn't picked it up. I didn't want to report this as a bug just in case its still something unique to us.

> "Double counting savings plan purchases in Power BI reports
> 
> One issue that has been reported more over the past month or so is the inconsistent EffectiveCost number in FOCUS data compared to amortized cost in the Azure portal. The reason for this is due to a bug in Cost Management where savings plan purchases are tracked as amortized (effective) cost. While we wait for the underlying issue to get resolved in Cost Management data, we’ve updated FinOps hubs and Power BI reports in 0.7 to account for this by removing the invalid effective cost values. You’ll see this in storage-based reports using Cost Management exports or FinOps hubs with storage as well as KQL queries and reports when using FinOps hubs with Data Explorer."

<img width="1229" alt="Screenshot 2024-12-31 at 8 59 15 AM" src="https://github.com/user-attachments/assets/ed2597da-a472-471f-8331-fab78866bddc" />

<img width="585" alt="Screenshot 2024-12-31 at 8 59 40 AM" src="https://github.com/user-attachments/assets/833739ed-0441-4e0e-ba1c-1cb7cb998c6c" />

**Comment by lundejd** [comment, marked as answer]
Strange I just noticed the 7.1 update and that it seems to have corrected this, though I deployed post December 9 and seemed to still see EffectiveCost as $0. I will double check but all of this is probably what you caught in that update. I just didn't see the update listed in this Discussion area and other than the changelog I'm not seeing the updated 7.1 materials so my guess is I re-deployed 7.0.

In any case looks like this was fixed but perhaps not deployed to us yet.

**Reply by flanakin** [reply]
Yes, we had a bug in the query that was excluding usage charges 😓 We fixed that in 0.7 update 1.

Sorry for the inconvenience and confusion this caused. Glad to hear things are working now.

Let us know if you run into any other issues.
